core/src/commonMain/kotlin/com/powersync/attachments/storage/IOLocalStorageAdapter.kt

```diff
@@ -0,0 +1,117 @@
+package com.powersync.attachments.storage
+
+import com.powersync.attachments.LocalStorage
+import com.powersync.db.runWrappedSuspending
+import io.ktor.utils.io.core.remaining
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.withContext
+import kotlinx.io.Buffer
+import kotlinx.io.buffered
+import kotlinx.io.files.FileSystem
+import kotlinx.io.files.Path
+import kotlinx.io.files.SystemFileSystem
+import kotlinx.io.readByteArray
+
+/**
+ * Storage adapter for local storage using the KotlinX IO library
+ */
+public class IOLocalStorageAdapter : LocalStorage {
+    private val fileSystem: FileSystem = SystemFileSystem
```

Comment:
I think we can now make this configurable too:

```suggestion
public class IOLocalStorageAdapter(
    private val fileSystem: FileSystem = SystemFileSystem
) : LocalStorage {
```

Reply:
👍 Implemented